### PR TITLE
Persist /runs query changes in the URL

### PR DESCRIPTION
**Pull request by lukebjerring**
## Description
More builder work (https://github.com/web-platform-tests/wpt.fyi/issues/535)

When the query is submitted on /runs tab, persist it to the URL.

## Review Information
- Visit the /runs page
- Change the query, submit
- See the changes in the URL
- Reload the page, see the same results

### CODE REVIEW

**Comment by mdittmer** [comment]
When trying the staging instance, I noticed the back button doesn't update the list of runs to the former state.

File: webapp/components/self-navigator.html [hunk 1]
```diff
@@ -84,9 +84,10 @@
 
       /**
        * Get query params to persist when creating history.
+       * Defaults to the queryParams property.
        */
       navigationQueryParams() {
-        return {};
+        return Object.assign({}, this.queryParams);
```
Comment (mdittmer) [inline webapp/components/self-navigator.html]: Just double-checking: Is `queryParams` a shallow object? Otherwise we may get some wonkiness here.

Comment (lukebjerring) [reply webapp/components/self-navigator.html]: Done.

**Comment by lukebjerring** [comment]
Back-button is broken for the /results/ tab too, so I'll fix that issue separately.

**Pull request merged**